verifier/check_known_issues.py

```diff
@@ -0,0 +1,152 @@
+# Functions to handle Known Issue category of results
+
+
+from report_template import reportTemplate
+
+from collections import defaultdict
+
+from difflib import HtmlDiff
+from difflib import Differ
+from difflib import SequenceMatcher
+
+from datetime import datetime
+
+import glob
+import json
+import logging
+import logging.config
+import os
+from string import Template
+import sys
+
+import unicodedata
+
+# Handle known issues database
+
+# Automatically compute some known issues with certain patterns of differences
+# in actual output vs. expected
+
+# E.g., NBSP vs SP in NodeJS DateTime in ICU73, ...
+
+class known_issue():
+    # Stores information on a single known issue.
+    # Ths may be stored in file or database
+    def __init__(self):
+        self.description = None
+        self.url_ref = None
+        self.test_type = None
+        self.instances = []
+        self.filter_fn = None
+        self.date_posted = None
+
+        self.issue_list = []
+
+# Constants
+NBSP = '\u202f'
+SP = '\u0020'
+
+# Global KI Info.
+# TODO: add this to a class
+known_issue_nbsp_sp = 'ASCII Space instead of NBSP'
+known_issue_replaced_numerals = 'Not creating non-ASCII numerals'
+
+# TODO! Load known issues from file of known problems rather than hardcoding
+
+class compute_known_issue():
+    def __init__(self):
+        return
+
+    def diff_nbsp_vs_sp(self, actual, expected):
+        # Returns true the only difference in the two strings
+        # is NBSP in expected vs. ASCII space in the actual result
+        if not expected or not actual:
+            return None
+
+        if expected.replace(NBSP, SP) == actual:
+            return True
+        return None
+
+    def numerals_replaced(self, expected, actual):
+        # If the only difference are one type of digit
+        # where other digits were expected, return True
+        sm = SequenceMatcher(None, expected, actual)
+        sm_opcodes = sm.get_opcodes()
+
+        digit_replace = False
+        non_digit_replacement = False
+        for diff in sm_opcodes:
+            kind = diff[0]
+            old_val = expected[diff[1]:diff[2]]
+            new_val = actual[diff[1]:diff[2]]
```

Comment:
please fix indices according to how the 4 indices contained within the 5-tuples returned by `get_opcodes` correspond to the 2 input strings that you originally provide as input:
https://docs.python.org/3/library/difflib.html#difflib.SequenceMatcher.get_opcodes

Reply:
Added text to clarify.